components/azure.key.manager/src/main/java/org/wso2/azure/client/AzureADRequestInterceptor.java

```diff
@@ -0,0 +1,25 @@
+package org.wso2.azure.client;
+
+import org.wso2.carbon.apimgt.impl.APIConstants;
+import org.wso2.carbon.apimgt.impl.recommendationmgt.AccessTokenGenerator;
```

Comment:
Looking at the implementation of this class, I guess we can move this to a common location @tharindu1st ?
Getting a class to an azure ad implementation from APIM recommendation feature is confusing and also it can break easily if someone do any change to the class related to the recommendation feature.

I guess the best thing to do for now is, copying this class into this repo itself and use it.

Reply:
No action items taken to this point

Reply:
@malinthaprasan +1